gateway/src/resty/concurrent/timer_task.lua

```diff
@@ -60,25 +56,32 @@ end
 
 local run_periodic, schedule_next, timer_execute
 
-run_periodic = function(self, run_now)
-  if not _M.task_is_active(self.id) then return end
+run_periodic = function(run_now, id, func, args, interval)
+  local t_start = ngx_now()
+
+  if not _M.task_is_active(id) then return end
 
   if run_now then
-    self.task(unpack(self.args))
+    func(unpack(args))
   end
 
-  schedule_next(self)
+  local t_diff = ngx_now() - t_start
+
+  local delay_for_next = interval - t_diff
```

Comment:
Is it really needed?
I think it is just fine to delay the execution. Documenting it is way easier than doing this. Also it could end up in situation when are two called immediately after each other because they took more than the delay.

Reply:
I think from a user perspective is more natural to expect a consistent reporting every X seconds. But there are no guarantees anyway because some reports might take more than the interval specified in the config. I don't think it's too important, we can remove it to simplify this a bit.

Reply:
I'd prefer it to be simpler. It can be documented pretty easily. 

For the record, `ngx.timer.every` behaves quite stupidly and will schedule timer every `N` seconds regardless how long the timer takes.